View/ViewHandler.php

```diff
@@ -502,4 +502,14 @@ private function getDataFromView(View $view)
 
         return $form;
     }
+
+    /**
+     * Resets internal object state at the end of the request
+     */
+    public function reset()
+    {
+        $this->exclusionStrategyGroups = [];
+        $this->exclusionStrategyVersion = null;
+        $this->serializeNullStrategy = null;
```

Comment:
I'm not sure resetting these fields is legit, they're defined in config so not per request, right ?

Reply:
Ok but then we should reset to config values, don't you think ?

Reply:
Well, we reset them to the default values of ViewHandler, like they would be at the start of a new request. If there is config to set them to something else it will be called and explicitly set them, otherwise they should be the same as what the class sets as default.

Reply:
To be more clear, the main difference between PPM and the normal PHP execution model is that objects are not destroyed at the end of the request. Otherwise the normal symfony request-response flow is executed (so if there is config to explicitly set values in the ViewHandler it will be called normally).

The problem only arises when the initial state of the object differs from what you would expect at the start of a normal request, as is the case here. So we just need to reset the class to what it would be at the start of a normal PHP request, before it even reaches Symfony.

Reply:
Thing is the initial state of the object differs from its state at the beginning of requests because of https://github.com/FriendsOfSymfony/FOSRestBundle/blob/master/DependencyInjection/FOSRestExtension.php#L400.

Reply:
OK, `setSerializeNullStrategy` was a bad example and seems to be explicitly set in all cases :)

But take for example `setExclusionStrategyGroups` a few lines above. If `$config['serializer']['groups'] is set in one request, but in the next request its empty, then ViewHandler will carry over the exclusionStrategyGroups attribute from the last request that set it, instead of being empty as it should.

Reply:
Sure but I think we should save differently the config settings in order to use them when resetting

Reply:
The thing is that technically the user can also call the `set*` methods in a controller, a listener, or other code regardless of the config. It could probably also work if `FOSRestExtension` explicitly calls all the `set*` methods of the `ViewHandler` regardless if they are empty (I think this is what you suggest?), but in my opinion it makes more sense to reset the object to its original state and call only the setters that are needed in `FOSRestExtension`.

Reply:
To be clearer, config options should be passed to the constructor and stored to be later restored. The constructor has already many arguments so we could group most of them under an `$options` argument but I wouldn't mind if you did just add the resetted config parameters as arguments in this PR.

Reply:
@GuilhemN its done, I have added the config options as a constructor argument instead of method calls and reset resets them to the original values at construction.